### ISSUE TITLE
Make the worker specific task queues work (this.cancelled undefined)

### ISSUE THREAD

**Issue by giuliohome** [temporalio/samples-typescript#310]
Please [refer to my comment here](https://github.com/temporalio/samples-typescript/issues/309#issuecomment-1817974209).

https://github.com/temporalio/samples-typescript/blob/98853010a11fc3c8a3dda49c7f0cdaa0e134fb32/worker-specific-task-queues/src/activities.ts#L17

**Comment by giuliohome** [comment]
It does work when I transform it into all sync without async activities (without importing sleep etc... see below)


```ts
export function createActivitiesForSameWorker() {
  return {
    async downloadFileToWorkerFileSystem(url: string, path: string): Promise<void> {
      const { log } = Context.current();
      log.info('Downloading and saving', { url, path });
      // Here's where the real download code goes
      const body = 'downloaded body';
      // await sleep(3000);
      syncfs.writeFileSync(path, body);
    },
    async workOnFileInWorkerFileSystem(path: string): Promise<void> {
      const { log } = Context.current();
      const content = await fs.readFile(path);
      const checksum = createHash('md5').update(content).digest('hex');
      // await sleep(3000);
      log.info('Did some work', { path, checksum });
    },
    async cleanupFileFromWorkerFileSystem(path: string): Promise<void> {
      const { log } = Context.current();
      // await sleep(3000);
      log.info('Cleaning up temp file', { path });
      await fs.rm(path);
    },
  };
}
```

**Comment by giuliohome** [comment]
Using the following delay instead of the above sleep solves the issue

```
function delay(ms: number) {
    return new Promise( resolve => setTimeout(resolve, ms) );
}
```

**Comment by giuliohome** [comment]
Will be solved with next release of sdk-typescript by making sleep an arrow function.

**Comment by mjameswh** [comment]
Resolved in [TS SDK 1.9.0](https://github.com/temporalio/sdk-typescript/releases/tag/v1.9.0).
